pkg/describe/describe.go

```diff
@@ -5,48 +5,59 @@ import (
 )
 
-func DescribePackage(
-	ctx context.Context,
-	pkgName string,
-) (*v1alpha1.Package, *client.PackageStatus, *v1alpha1.PackageManifest, string, error) {
-	pkgClient := client.FromContext(ctx)
+func DescribeInstalledPackage(ctx context.Context, pkgName string) (
+	*v1alpha1.Package, *v1alpha1.PackageManifest, error) {
+
+	pkgClient := cliutils.PackageClient(ctx)
+	repoClient := cliutils.RepositoryClientset(ctx)
 	var pkg v1alpha1.Package
-	var status *client.PackageStatus
-	if err := pkgClient.Packages().Get(ctx, pkgName, &pkg); err == nil {
-		// pkg installed: try to use installed manifest
-		status = client.GetStatusOrPending(&pkg.Status)
-		if installedManifest, err := manifest.GetInstalledManifestForPackage(ctx, pkg); err == nil {
-			return &pkg, status, installedManifest, "", nil
-		} else if !(errors.Is(err, manifest.ErrPackageNoManifest) ||
-			errors.Is(err, manifest.ErrPackageNoOwnedPackageInfo) ||
-			apierrors.IsNotFound(err)) {
-			return nil, nil, nil, "", err
-		}
-	} else if !apierrors.IsNotFound(err) {
-		return nil, nil, nil, "", err
+	err := pkgClient.Packages().Get(ctx, pkgName, &pkg)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if installedManifest, err := manifest.GetInstalledManifestForPackage(ctx, pkg); err == nil {
+		return &pkg, installedManifest, nil
+	} else if !errors.Is(err, manifest.ErrPackageNoManifest) {
+		return nil, nil, err
 	}
 
+	// pkg is installed, but has either no manifest or owned package info (yet): use manifest in this version from repo
 	var packageManifest v1alpha1.PackageManifest
-	if pkg.Spec.PackageInfo.Version != "" {
-		// pkg is installed, but has either no manifest or owned package info (yet): use manifest in this version from repo
-		if err := repo.FetchPackageManifest("", pkgName, pkg.Spec.PackageInfo.Version, &packageManifest); err != nil {
-			return nil, nil, nil, "", err
-		} else {
-			return &pkg, status, &packageManifest, "", nil
-		}
+	err = repoClient.ForPackage(pkg).FetchPackageManifest(pkgName, pkg.Spec.PackageInfo.Version, &packageManifest)
+	if err != nil {
+		return nil, nil, err
 	} else {
-		// pkg not installed: use latest manifest from repo
-		// TODO: Returning latestVersion in this way seems weird. We should find a better way.
-		if latestVersion, err := repo.FetchLatestPackageManifest("", pkgName, &packageManifest); err != nil {
-			return nil, nil, nil, "", err
+		return &pkg, &packageManifest, nil
+	}
+}
+
+func DescribeLatestVersion(ctx context.Context, repositoryName string, packageName string) (
+	*v1alpha1.PackageManifest, string, error) {
+
+	repoClient := cliutils.RepositoryClientset(ctx)
+	if len(repositoryName) == 0 {
+		if repos, err := repoClient.Aggregate().GetReposForPackage(packageName); err != nil {
+			return nil, "", err
 		} else {
-			return nil, nil, &packageManifest, latestVersion, nil
+			for _, repo := range repos {
+				repositoryName = repo.Name
+				if !repoclient.IsDefaultRepository(repo) {
```

Comment:
I'm not sure… the spec says that it should use the first repository (sorted alphabetically by name) that is not the default repository. I guess it depends on the usage which would make more sense. My initial though was that if someone has an additional repo where there is a package with the same name as the default repository, they probably want to override the default. On the other hand, it could be argued that "overrides" should be the default anyways :shrug: